libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultTracer.java

```diff
@@ -97,4 +100,10 @@ protected void addDefaultAttributes(Span span) {
         span.addAttribute(THREAD_NAME, Thread.currentThread().getName());
     }
 
+    @Override
+    public SpanScope startSpan(String spanName, Map<String, List<String>> header, Attributes attributes) {
```

Comment:
Super minor, but we have many headers here:

```suggestion
    public SpanScope startSpan(String spanName, Map<String, List<String>> headers, Attributes attributes) {
```